### DISCUSSION TITLE
Warnings system

### DISCUSSION THREAD

**Discussion by enrico-ghidoni** [CntoDev/level-up#83] [Ideas]
There are several warnings to be shown to both Interviewers and Member Managers.

 - when a member hasn't attended enough events (2 events in the past month)
 - when a recruit's 8 week probation phase is close to ending
 - when a recruit is close to not fulfilling their mod assessment (due 2 weeks from joining)

I'm sure there are going to be many other warnings that are either already in the `django-roster` or that could be useful to introduce. But I would stop with the three listed above for the MVP.

For the UX bit, different roles should see different warnings and only relative to their branch. For example, Interviewers should not be able to see warnings for Member Managers and viceversa. Also, displaying the warnings on the frontend should allow for some basic search and filtering features.

@milivojm if you agree this is enough for the MVP I can go ahead and split this into different sub-tasks.

**Comment by milivojm** [comment]
_Warnings_ concept is obsolete. If we automate, we automate. If not, we can do this in Excel.

**Comment by enrico-ghidoni** [comment]
Not really. As I mentioned there are a bunch of different warnings for different things (i.e. contributor tags) and as explained in #62 I do believe that we should build automation at a later stage and keeping the system as self-explanatory as possible. So even if we end up automating things, a "notice board" should be there to inform staff about what has been automatically done and why

**Comment by enrico-ghidoni** [comment]
Moving discussion about automation to #68

**Comment by milivojm** [comment]
I've automated warning conditions described above, you can close it.

**Comment by milivojm** [comment]
Do we need this for MVP? Or at all?

**Comment by milivojm** [comment]
Warnings:
- ApplicationFormSubmitted - interviewers
- EnoughEventsAttended - interviewers
- RecruitAssessmentExpired - interviewers
- RecruitTrialExpired - interviewers

Events or Commands? - ask @JamesTheClarke @Gikerl 
- PromoteMember
- DemoteMember
- DischargeMember

**Comment by milivojm** [comment]
Broke down this epic into smaller issues -> https://github.com/CntoDev/rooster/milestone/5, we can lock this topic.
